plugins/catalog-backend-module-msgraph/src/microsoftGraph/config.ts

```diff
@@ -49,12 +48,14 @@ export type MicrosoftGraphProviderConfig = {
   tenantId: string;
   /**
    * The OAuth client ID to use for authenticating requests.
+   * If specified, ClientSecret must also be specified
    */
-  clientId: string;
+  clientId?: string;
   /**
    * The OAuth client secret to use for authenticating requests.
+   * If specified, ClientId must also be specified
    */
-  clientSecret: string;
+  clientSecret?: string;
```

Comment:
these two changes will have to be reflected in the `config.d.ts` file in the root of the same plugin - in three different places actually, since it is a bit weirdly structured :)